src/main/java/com/mindee/MindeeClient.java

```diff
@@ -25,209 +25,262 @@ public class MindeeClient {
   private final MindeeApi mindeeApi;
   private final PdfOperation pdfOperation;
 
-  public MindeeClient(PdfOperation pdfOperation, MindeeApi mindeeApi) {
-    this.pdfOperation = pdfOperation;
-    this.mindeeApi = mindeeApi;
+  /**
+   * Create a default MindeeClient.
+   * You'll need to set the API key in the environment for this approach to work properly.
+   */
+  public MindeeClient() {
+    this.pdfOperation = new PdfBoxApi();
+    this.mindeeApi = createDefaultApi("");
   }
 
-  public LocalInputSource loadDocument(
-      InputStream fileStream,
-      String fileName
-  ) throws IOException {
-    return new LocalInputSource(fileStream, fileName);
+  /**
+   * Create a default MindeeClient.
+   *
+   * @param apiKey The api key to use.
+   */
+  public MindeeClient(String apiKey) {
+    this.pdfOperation = new PdfBoxApi();
+    this.mindeeApi = createDefaultApi(apiKey);
   }
 
-  public LocalInputSource loadDocument(
-      byte[] fileAsByteArray,
-      String filename
-  ) {
-    return new LocalInputSource(fileAsByteArray, filename);
+  /**
+   * Create a MindeeClient using a MindeeApi.
+   *
+   * @param mindeeApi The MindeeApi implementation to be used by the created MindeeClient.
+   */
+  public MindeeClient(MindeeApi mindeeApi) {
+    this.pdfOperation = new PdfBoxApi();
+    this.mindeeApi = mindeeApi;
   }
 
-  public LocalInputSource loadDocument(File file) throws IOException {
-    return new LocalInputSource(file);
+  /**
+   * Create a MindeeClient.
+   * @param pdfOperation
+   * @param mindeeApi
+   */
+  public MindeeClient(PdfOperation pdfOperation, MindeeApi mindeeApi) {
+    this.pdfOperation = pdfOperation;
+    this.mindeeApi = mindeeApi;
   }
 
-  public LocalInputSource loadDocument(
-      String fileInBase64Code,
-      String filename
-  ) {
-    return new LocalInputSource(fileInBase64Code, filename);
+  private static MindeeApi createDefaultApi(String apiKey) {
+    MindeeSettings mindeeSettings;
+    if (apiKey != null && !apiKey.trim().isEmpty()) {
+      mindeeSettings = new MindeeSettings(apiKey);
+    } else {
+      mindeeSettings = new MindeeSettings();
+    }
+    return MindeeHttpApi.builder()
+      .mindeeSettings(mindeeSettings)
+      .build();
   }
 
-  public <T extends Inference> PredictResponse<T> parseQueued(Class<T> type, String jobId) {
-    return this.mindeeApi.checkJobStatus(type, jobId);
+  public <T extends Inference> AsyncPredictResponse<T> parseQueued(Class<T> type, String jobId) {
+    return this.mindeeApi.documentQueueGet(
+        type,
+        new Endpoint(type),
```

Comment:
Minor - we could consider using parameterized types for Endpoint to ensure correct ednpoints are used to avoid something like 

this.mindeeApi.documentQueueGet(
        type,
        new Endpoint(I**nvoiceV4.class**),

Reply:
This should not happen so long as the users go through the provided functions... let's keep as is for now.